src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassSortServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package edu.cornell.kfs.rass.batch.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import edu.cornell.kfs.rass.batch.service.RassSortService;
+import edu.cornell.kfs.rass.batch.xml.RassXmlAgencyEntry;
+
+public class RassSortServiceImpl implements RassSortService {
+    private static final Logger LOG = LogManager.getLogger(RassSortServiceImpl.class);
+
+    @Override
+    public List<RassXmlAgencyEntry> sortRassXmlAgencyEntriesForUpdate(List<RassXmlAgencyEntry> agencies) {
+        AgenciesSortHelper agencySortHelper = new AgenciesSortHelper();
+        sortAgenciesWithNoReportsToAgency(agencies, agencySortHelper);
+        sortAgenciesWithReportsToAgencyThatHasBeenSorted(agencySortHelper);
+        sortLeftOverAgencies(agencySortHelper);
+        return agencySortHelper.sortedAgencies;
+    }
+
+    private void sortAgenciesWithNoReportsToAgency(List<RassXmlAgencyEntry> agencies, AgenciesSortHelper agencySortHelper) {
+        for (RassXmlAgencyEntry entry : agencies) {
+            if (StringUtils.isBlank(entry.getReportsToAgencyNumber())) {
+                agencySortHelper.agencyNumberInSortedList.add(entry.getNumber());
+                agencySortHelper.sortedAgencies.add(entry);
+            } else {
+                agencySortHelper.leftOverAgencies.add(entry);
```

Comment:
are the left over agencies in fact the unsorted ones? I am wondering if there is a better name for what goes in this list